packages/pharos/src/components/coach-mark/pharos-coach-mark.ts

```diff
@@ -0,0 +1,151 @@
+import { coachMarkStyles } from './pharos-coach-mark.css';
+import ScopedRegistryMixin from '../../utils/mixins/scoped-registry';
+import debounce from '../../utils/debounce';
+import { computePosition, flip, shift, offset } from '@floating-ui/dom';
+
+export type Side = 'top' | 'right' | 'bottom' | 'left';
+export type Alignment = 'start' | 'center' | 'end';
+export type Delay = 'none' | 'short' | 'long';
+export type Variant = 'light' | 'dark';
+
+/**
+ * Pharos coach mark component.
+ *
+ * @tag pharos-coach-mark
+ *
+ */
+export class PharosCoachMark extends ScopedRegistryMixin(PharosElement) {
+  static elementDefinitions = {
+    'pharos-button': PharosButton,
+    'pharos-heading': PharosHeading,
+  };
+
+  public static override get styles(): CSSResultArray {
+    return [coachMarkStyles];
+  }
+
+  /**
+   * Indicates that the coach mark should not be displayed on the page
+   * @attr hide
+   */
+  @property({ type: Boolean, reflect: true })
+  public hide = true;
+
+  /**
+   * Indicates which side of the base element the coachmark should appear on
+   * @attr side
+   * @type {Side}
+   */
+  @property({ reflect: true })
+  public side: Side = 'bottom';
+
+  /**
+   * Indicates how the coach mark carat should be aligned in relation to the coach mark content
+   * @attr alignment
+   * @type {Alignment}
+   */
+  @property({ reflect: true })
+  public alignment: Alignment = 'center';
+
+  /**
+   * Dictates how long to wait between coach mark trigger and start of coach mark fade in animation
+   * @attr delay
+   * @type {Delay}
+   */
+  @property({ reflect: true })
+  public delay: Delay = 'short';
+
+  /**
+   * Text content for the coach mark header
+   * @attr header
+   * @type {String}
+   */
+  @property({ reflect: true })
+  public header = '';
+
+  /**
+   * Style variant
+   * @attr variant
+   * @type {Variant}
+   */
+  @property({ reflect: true })
+  public variant: Variant = 'dark';
+
+  /**
+   * Set minimum width of coach mark
+   * @attr width
+   * @type {number}
+   */
+  @property({ reflect: true })
+  public width = 250;
+
+  constructor() {
+    super();
+    this.setOffset();
+    this.resizeObserver.observe(document.body);
+  }
+
+  private resizeObserver = new ResizeObserver(
+    debounce((entries) => {
+      entries.map(() => {
+        this.setOffset();
+        this.requestUpdate();
+      });
+    }, 100)
+  );
+
+  private setOffset() {
+    const id: string = this.getAttribute('id') || '';
+    const targetElement: Element | null = document.querySelector(`[data-coach-mark="${id}"]`);
+    if (!targetElement) return;
+
+    computePosition(targetElement, this, {
+      placement: this.side,
+      middleware: [flip(), shift({ padding: 10 }), offset(20)],
+    }).then(({ x, y }) => {
+      Object.assign(this.style, {
+        left: `${x}px`,
+        top: `${y}px`,
+      });
+    });
+  }
+
+  protected override render(): TemplateResult {
+    this.setOffset();
```

Comment:
does this need called at render if it's already called in the constructor?

Reply:
I think the component may get constructed just once, but rendered multiple times?